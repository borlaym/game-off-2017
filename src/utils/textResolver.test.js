import resolveText from './textResolver';

const maleCharacter = {
	sex: 'male',
	name: 'Joe',
	tags: ['PILOT', 'ENGINEER'].map(_ => ({ id: _ })),
};

const femaleCharacter = {
	sex: 'female',
	name: 'Sarah',
	tags: ['OUTLAW'].map(_ => ({ id: _ })),
};

describe('resolve text', () => {
	it('Resolves XE, XYR, XYRS', () => {
		const result = resolveText({
			template: 'Should XE escape, we will be on XYRS tail. Keep XYR locked up.',
			character: maleCharacter,
		});
		expect(result).toMatchSnapshot();
		const result2 = resolveText({
			template: 'Should XE escape, we will be on XYRS tail. Keep XYR locked up.',
			character: femaleCharacter,
		});
		expect(result2).toMatchSnapshot();
	});
	it('Resolves NAME', () => {
		const result = resolveText({
			template: 'NAME is a dangerous individual.',
			character: maleCharacter,
		});
		expect(result).toMatchSnapshot();
	});
	it('Resolves template strings', () => {
		const template = '{% if (o.character.tags.PILOT) { %}Piloting was great for XYR.{% } else { %}Piloting was not XYRS thing.{% } %}';
		const result = resolveText({
			template,
			character: maleCharacter,
		});
		expect(result).toMatchSnapshot();
		const result2 = resolveText({
			template,
			character: femaleCharacter,
		});
		expect(result2).toMatchSnapshot();
	});
});
