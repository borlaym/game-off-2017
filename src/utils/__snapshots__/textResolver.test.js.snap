// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolve text Resolves NAME 1`] = `"Joe is a dangerous individual."`;

exports[`resolve text Resolves XE, XYR, XYRS 1`] = `"Should he escape, we will be on his tail. Keep him locked up."`;

exports[`resolve text Resolves XE, XYR, XYRS 2`] = `"Should she escape, we will be on her tail. Keep her locked up."`;

exports[`resolve text Resolves template strings 1`] = `"Piloting was great for him."`;

exports[`resolve text Resolves template strings 2`] = `"Piloting was not her thing."`;
